### PR TITLE
Improve description of Julia keywords

### PR DESCRIPTION
**Pull request by bkamins**
Clarification of keyword arguments documentation following #29197.

### CODE REVIEW

File: doc/src/base/base.md [hunk 1]
```diff
@@ -37,6 +37,21 @@ ans
...
+`return`, `struct`, `true`, `try`, `using`, `while`.
+Those keywords are not allowed to be used as variable names.
+
+The follwoing two word sequences are reserved:
+`abstract type`, `mutable struct`, `primitive type`.
+However, you can create variables with names:
+`abstract`, `mutable`, `primitive` and `type`.
+
+Finally `where` is resolved contextually as a keyword in parametric method
+definition syntax, but creation of variable named `where` is allowed.
```
Comment (JeffBezanson) [inline doc/src/base/base.md]: `where` is parsed as an infix operator for writing parametric method definitions and certain types. `in` and `isa` are also parsed as infix operators. However these may generally be used as variable names.

**Review by ararslan — APPROVED** [review]

File: doc/src/base/base.md [hunk 2]
```diff
@@ -37,6 +37,22 @@ ans
...
 
+This is the list of reserved keywords in Julia:
+`baremodule`, `begin`, `break`, `catch`, `const`, `continue`, `do`,
+`else`, `elseif`, `end`, `export`, `false`, `finally`, `for`, `function`,
+`global`, `if`, `import`, `let`, `local`, `macro`, `module`, `quote`,
+`return`, `struct`, `true`, `try`, `using`, `while`.
+Those keywords are not allowed to be used as variable names.
+
+The following two word sequences are reserved:
+`abstract type`, `mutable struct`, `primitive type`.
```
Comment (PallHaraldsson) [inline doc/src/base/base.md]: three word sequences

Comment (KristofferC) [reply doc/src/base/base.md]: It is better to just open a new PR with the suggested changes instead of commenting on old merged PRs.

Comment (bkamins) [reply doc/src/base/base.md]: Thank you. I will make a PR to clarify it.

**Pull request merged**